### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by BogdanParubchenk01**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,35 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
```
Comment (DarkMistyRoom) [inline src/transformState.js:8]:
```suggestion
  for (const action of actions) {
```

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,35 @@
...
+    for (const key in obj) {
```
Comment (DarkMistyRoom) [inline src/transformState.js:9]: you don't need a second for here. an action is an object and you know all its properties. you just need to check `action.type`

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,35 @@
...
+            }
+          }
+        }
+      }
+
+      if (obj[key] === 'clear') {
+        for (const result in state) {
+          delete state[result];
+        }
+      }
```
Comment (DarkMistyRoom) [inline src/transformState.js:32]: Use switch statement if you have limited amount of conditions.

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 4]
```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const push in action.extraData) {
```
Comment (IvanFesenko) [inline src/transformState.js:11]: push better call value.

**Review by IvanFesenko — APPROVED** [review]
Well done

**Pull request closed without merging**